packages/terra-form-select/src/multiple/Menu.jsx

```diff
@@ -341,40 +338,25 @@ class Menu extends React.Component {
 
       const option = MenuUtil.findByValue(children, active);
       // Handles communicating the case where a regular option is selected to screen readers.
-      /*
-        Detecting if browser is not Safari before updating aria-live as there is some odd behaivor
-        with VoiceOver on desktop, that causes the selected option to be read twice when this is
-        is added to aria-live container.
-        When we shift focus back to select, VoiceOver automatically reads the display text.
-        Using aria-hidden on the display does not prevent VO from reading the display text and so it
-        results in reading the display text followed by reading the aria-live message which is
-        the display text + 'selected'
-        */
-      if (SharedUtil.isSafari()) {
-        this.props.visuallyHiddenComponent.current.innerText = `${option.props.display} ${selectedTxt}`;
-      } else {
-        this.props.visuallyHiddenComponent.current.innerText = `${option.props.display} ${selectedTxt}`;
+      if (visuallyHiddenComponent && visuallyHiddenComponent.current) {
+        visuallyHiddenComponent.current.innerHTML = intl.formatMessage({ id: 'Terra.form.select.selectedText' }, { text: option.props.display });
+      }
+
+      if (onSelect) {
+        onSelect(option.props.value, option);
       }
-      onSelect(option.props.value, option);
     } else if (keyCode === KeyCode.KEY_RETURN && active !== null && MenuUtil.includes(value, active)) {
       event.preventDefault();
+
       const option = MenuUtil.findByValue(children, active);
       // Handles communicating the case where a regular option is Unselected to screen readers.
-      /*
-        Detecting if browser is not Safari before updating aria-live as there is some odd behaivor
-        with VoiceOver on desktop, that causes the selected option to be read twice when this is
-        is added to aria-live container.
-        When we shift focus back to select, VoiceOver automatically reads the display text.
-        Using aria-hidden on the display does not prevent VO from reading the display text and so it
-        results in reading the display text followed by reading the aria-live message which is
-        the display text + 'Unselected'
-        */
-      if (SharedUtil.isSafari()) {
-        this.props.visuallyHiddenComponent.current.innerText = `${option.props.display} ${unselectedTxt}`;
-      } else {
-        this.props.visuallyHiddenComponent.current.innerText = `${option.props.display} ${unselectedTxt}`;
+      if (visuallyHiddenComponent && visuallyHiddenComponent.current) {
+        visuallyHiddenComponent.current.innerHTML = intl.formatMessage({ id: 'Terra.form.select.unselectedText' }, { text: option.props.display });
+      }
+
+      if (onDeselect) {
```

Comment:
Now that the frames are variant specific these callback functions (onSelect / onDeselect) could probably be marked as required. This would remove a lot of the unnecessary guard checks. The Menu should be a private implementation but if you're concerned about passivity this could be a change for another time. But the callback functions should be considered guaranteed here, even if they are not marked as required.

Reply:
True, they are guaranteed to be here, given that we always pass `handleX` from above

It ain't public, so I'm fine with making them `.isRequired`

Reply:
Actually, this isn't entirely true..

https://github.com/cerner/terra-core/blob/master/packages/terra-form-select/src/multiple/Frame.jsx#L582-L594

`onDeselect` is passed from `props`, which has a defaultValue of `undefined` [here](https://github.com/cerner/terra-core/blob/master/packages/terra-form-select/src/multiple/Frame.jsx#L106)

So, in order to be confident that they are present, I say we either:

- [ ] add defaultProps of `() => {}` for all callbacks
- [ ] keep truthy checks for callbacks
- [ ] add bound wrapper methods in `Frame` (later, the individual variant implementations)

The last option is similar to the first, but would end up basically adding more noise for the same effect, with maybe a lil more safety e.g.:

```jsx
handleOnDeselect = (value) => {
  const { onDeselect } = this.props;

  if (onDeselect) {
    onDeselect(value);
  }
}
```

Reply:
> We can tackle it in another issue. 
:+1: I'd leave the code as-is. We can talk about it in our weekly sync but I don't want to derail this PR.

Reply:
These functions are guaranteed to exist from the base Multiple component.

https://github.com/cerner/terra-core/blob/master/packages/terra-form-select/src/MultiSelect.jsx#L203-L204

Reply:
This can/will be address when the Frame and Selects are flattened then.

Reply:
When the frame is flattened this will become more obvious